File: backend-java/src/main/java/com/aiproject/article/dto/ArticleResponse.java
package com.aiproject.article.dto;

import com.aiproject.article.Article;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.Instant;

@Data
@AllArgsConstructor
public class ArticleResponse {

    private Long id;
    private String title;
    private String content;
    private Long authorId;
    private String author;
    private boolean published;
    private Instant createdAt;
    private Instant updatedAt;

    public static ArticleResponse from(Article article, String authorName) {
        return new ArticleResponse(
                article.getId(),
                article.getTitle(),
                article.getContent(),
                article.getUserId(),
                authorName,
                article.isPublished(),
                article.getCreatedAt(),
                article.getUpdatedAt());
    }
}

File: backend-dotnet/AiProject.Api/DTOs/ArticleResponse.cs
using AiProject.Api.Models;

namespace AiProject.Api.DTOs;

public class ArticleResponse
{
    public long Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public long AuthorId { get; set; }
    public string Author { get; set; } = string.Empty;
    public bool Published { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public static ArticleResponse From(Article article, string authorName) => new()
    {
        Id = article.Id,
        Title = article.Title,
        Content = article.Content,
        AuthorId = article.UserId,
        Author = authorName,
        Published = article.Published,
        CreatedAt = article.CreatedAt,
        UpdatedAt = article.UpdatedAt
    };
}
